drivers/sensor/adi/adxl367/adxl367_trigger.c

```diff
@@ -55,6 +57,10 @@ static void adxl367_gpio_callback(const struct device *dev,
 
 	gpio_pin_interrupt_configure_dt(&cfg->interrupt, GPIO_INT_DISABLE);
 
+	if (IS_ENABLED(CONFIG_ADXL367_STREAM)) {
+		adxl367_stream_irq_handler(drv_data->dev);
+	}
+
```

Comment:
They are processing different IRQs (stream is expecting FIFO full or FIFO watermark and trigger is expecting Data ready or Activity threshold) so if both are enabled it should work ok. If for example, Data ready is triggered, stream code would call RTIO to read STATUS register (when callback is triggered it would determine that IRQ is not for the stream and do nothing) and adxl367_thread_cb would do the processing of that IRQ. Similar thing would happen when FIFO full would trigger (stream would do the processing and adxl367_thread_cb would do nothing). This is just my opinion. I could be wrong.